nncf/experimental/tensor/numpy_functions.py

```diff
@@ -28,68 +28,72 @@
 DTYPE_MAP_REV = {v: k for k, v in DTYPE_MAP.items()}
 
 
-@functions.device.register(np.ndarray)
-@functions.device.register(np.number)
+def registry_numpy_types(singledispatch_fn):
+    """
+    Decorator to register function to singledispatch for numpy classes.
+
+    :param singledispatch_fn: singledispatch function.
+    """
+
+    def inner(func):
+        singledispatch_fn.register(np.ndarray)(func)
+        singledispatch_fn.register(np.generic)(func)
```

Comment:
This function refers to `np.generic`, while the type hint in the implementations is `Union[np.ndarray, np.number]`. Should establish consistency here.